### ISSUE TITLE
 A couple more Overlay Issues

### ISSUE THREAD

**Issue by GoogleCodeExporter** [gerhardol/gps-running#5]
```
What steps will reproduce the problem?
1.
2.
3.

What is the expected output? What do you see instead?


What version of the product are you using? On what operating system?


Please provide any additional information below.


I found a few more issues with the OverlayPlugin.

OverlayView.cs
getDataSeries:
  series.Points.Add(series.Points.Count,new PointF(x, y));
should be changed to:
  if ( series.Points.IndexOfKey( x ) == -1 )
    series.Points.Add( x, new PointF( x, y ) );

getCategoryAverage:
  average.Points.Add( index++, new PointF( x, y / seen ) );
should be changed to:
  if ( average.Points.IndexOfKey( x ) == -1 )
    average.Points.Add( x, new PointF( x, y / seen ) );

The above two fixes should resolve a SportTracks bug where the incorrect 
selected range average is shown (in a box on the y-axis).
SportTracks appears to be using the Key value to calculate the average.  
This fix is a workaround.

maBox_LostFocus:
The following line should be removed.
  maBox.Text = UnitUtil.Distance.ToString(value);
In the Catch the following change should be made:
  //maBox.Text = UnitUtil.Distance.ToString(Settings.MovingAverageLength);
  maBox.Text = UnitUtil.Elevation.ToString( Settings.MovingAverageLength );

In other words, maBox.Text should either stick with UnitUtil.Distance, or 
with UnitUtil.Elevation.
```

Original issue reported on code.google.com by `splani...@gmail.com` on 6 Mar 2010 at 7:42
